dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPIImpl.java

```diff
@@ -130,13 +128,46 @@ public Host resolveHostName(String serverName, User user, boolean respectFronten
                 hostCache.addHostAlias(serverName, host);
             }
         }
-        
-        if(APILocator.getPermissionAPI().doesUserHavePermission(host, PermissionAPI.PERMISSION_READ, user, respectFrontendRoles)){
-            return host;
-        } else {
+
+        checkHostPermission(user, respectFrontendRoles, host);
+        return host;
+    }
+
+    @Override
+    @CloseDBIfOpened
+    public Host resolveHostNameWithoutDefault(String serverName, User user, boolean respectFrontendRoles) throws DotDataException, DotSecurityException {
+
+        Host host = hostCache.getHostByAlias(serverName);
+        User systemUser = APILocator.systemUser();
+
+        if(host == null){
+            try {
+                host = findByNameNotDefault(serverName, systemUser, respectFrontendRoles);
+            } catch (Exception e) {
+                return findDefaultHost(systemUser, respectFrontendRoles);
```

Comment:
where are permissions validated here? the method is called using system user instead of the user sent as a parameter